cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1450,23 +1450,26 @@ class DataFrameBlockCreator {
     return Status::OK();
   }
 
-  Status WriteTableToBlocks(int nthreads) {
+  Status WriteTableToBlocks(bool use_threads) {
     auto WriteColumn = [this](int i) {
       std::shared_ptr<PandasBlock> block;
       RETURN_NOT_OK(this->GetBlock(i, &block));
       return block->Write(this->table_->column(i), i, this->column_block_placement_[i]);
     };
 
-    int num_tasks = table_->num_columns();
-    nthreads = std::min<int>(nthreads, num_tasks);
-    if (nthreads == 1) {
-      for (int i = 0; i < num_tasks; ++i) {
-        RETURN_NOT_OK(WriteColumn(i));
+    arrow::internal::ThreadPool* pool = arrow::internal::CPUThreadPool();
+    std::vector<std::future<Status>> futures;
+    for (int i = 0; i < table_->num_columns(); ++i) {
+      futures.push_back(pool->Submit(WriteColumn, i));
+    }
+    auto final_status = Status::OK();
+    for (auto& fut : futures) {
+      Status st = fut.get();
+      if (!st.ok()) {
+        final_status = st;
       }
-    } else {
-      RETURN_NOT_OK(ParallelFor(nthreads, num_tasks, WriteColumn));
     }
-    return Status::OK();
+    return final_status;
```

Comment:
Put this "merge all futures Status return values" into a helper function?

Reply:
Do you think this should be a static method on the Status class?